### DISCUSSION TITLE
Why does `open_gif` create bad quality images?

### DISCUSSION THREAD

**Discussion by stla** [pyvista/pyvista#1742] [Q&A] [answered]
Hello,

  When I take a png screenshot of my wavy cylinder, it is beautiful:

![aaaaaa](https://user-images.githubusercontent.com/4466543/138395629-9b5b2e8f-68b2-4fe6-9279-b1db86ba9daf.png)

  But when I make a gif with `open_gif` and `orbit_on_path`, I get bad quality:

![214a1m51](https://user-images.githubusercontent.com/4466543/138395727-e11275a2-2bca-4d5d-b4dc-a430873add56.gif)

  Why that? Instead of making a gif, is it possible to take a png screenshot of each frame? Then I could do a gif with ImageMagick or gifski.

```python
import tempfile
import os
from math import pi
import numpy as np
import pyvista as pv

# R is radius
# a is sinusoid amplitude
# θ is the parameter (angle), from 0 to 2π
# n is number of sinusoids 
# beta is the distortion parameter 
def F(R, a, n, beta):
    return lambda theta, h: np.array([
            (R + a * np.sin(n*theta)) * np.cos(theta + h**beta-1),
            (R + a * np.sin(n*theta)) * np.sin(theta + h**beta-1),
            h
        ])

n_theta = 150
n_h = 250
theta_ = np.linspace(0, 2*pi, n_theta)
hmin = 1
hmax = 20
h_ = np.linspace(hmin, hmax, n_h)

u, v = np.meshgrid(theta_, h_)
L = F(R=4, a=1, n=3, beta=0.6)
x, y, z = L(u, v)
grid = pv.StructuredGrid(x, y, z)
mesh = grid.extract_geometry().clean(tolerance=1e-6)
mesh.point_data["distance"] = np.linalg.norm(mesh.points, axis=1)

curve1 = np.empty((0,3))
curve2 = np.empty((0,3))
for theta in theta_:
    curve1 = np.vstack((curve1, L(theta, hmin)))
    curve2 = np.vstack((curve2, L(theta, hmax)))
spline1 = pv.Spline(curve1, 100)
tube1 = spline1.tube(radius=0.3).extract_geometry()
spline2 = pv.Spline(curve2, 100)
tube2 = spline2.tube(radius=0.3).extract_geometry()

# GIF:
filename = os.path.join(tempfile._get_default_tempdir(),
                        next(tempfile._get_candidate_names()) + ".gif")
pltr = pv.Plotter(window_size=[512, 512])
pltr.set_background("#363940")
_ = pltr.add_mesh(
    mesh, specular=20, pbr=True, metallic=10, cmap="turbo", diffuse=5,
    show_scalar_bar=False
)
_ = pltr.add_mesh(
    tube1, specular=20, pbr=True, metallic=10, color="yellow", diffuse=25
)
_ = pltr.add_mesh(
    tube2, specular=20, pbr=True, metallic=10, color="yellow", diffuse=25
)
viewup = [1, 0, 1]
pltr.camera.view_angle = 20
orbit = pltr.generate_orbital_path(factor=5, n_points=100,
                                      shift=1.5*mesh.length, viewup=viewup)
pltr.open_gif(filename)
pltr.orbit_on_path(orbit, write_frames=True, viewup=viewup, 
                      step=0.2)
pltr.close()


# PNG SCREENSHOT: 
pltr = pv.Plotter(window_size=[512,512], off_screen=True)
pltr.set_background("#363940")
pltr.add_mesh(
    mesh, specular=20, pbr=True, metallic=10, cmap="turbo", diffuse=5,
    show_scalar_bar=False
)
pltr.add_mesh(
    tube1, specular=20, pbr=True, metallic=10, color="yellow", diffuse=25
)
pltr.add_mesh(
    tube2, specular=20, pbr=True, metallic=10, color="yellow", diffuse=25
)
pltr.show(screenshot="WavyCylinder.png")
```

**Comment by banesullivan** [comment]
I think what you are experiencing is the fact that GIFs have a limit of 256 colors and they have trouble interpolating between colors. There is a neat example of the issue and possible solutions here: https://en.wikipedia.org/wiki/GIF#True_color

This is most notable when your scene has a color bar:

![optimized](https://user-images.githubusercontent.com/22067021/142668977-86c2ac0a-60c9-4f08-a5ac-3e35c5f433e7.gif)

ref https://docs.pyvista.org/examples/02-plot/gif.html

I don't know of any way around this. I'd recommend using `open_movie` instead:

```py
plotter.open_movie("wave.mov")
```

https://user-images.githubusercontent.com/22067021/142669093-b58e786f-c886-4d41-9e30-dcfb337a5e7c.mov

**Comment by stla** [comment]
I don't think this is the reason. Because when I do some animations without using `open_gif`, they are clean: see [here](https://gist.github.com/stla/078aef2b6fbcc3297f090e673bbe2c4d). So this is rather a problem with `open_gif` and/or `orbit_on_path`.

**Comment by camUrban** [comment, marked as answer]
@stla I came across a similar issue (see #2073) and developed a workaround. Hopefully, it helps you!
